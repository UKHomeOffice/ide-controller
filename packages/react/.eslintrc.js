// @todo add vs code format on save wiki/ readme on github
module.exports = {
  env: {
    browser: true,
    es2020: true,
    node: true,
    jest: true,
  },
  extends: ['plugin:react/recommended', 'airbnb', 'prettier'],
  parserOptions: {
    ecmaFeatures: {
      jsx: true,
    },
    ecmaVersion: 11,
    sourceType: 'module',
  },
  plugins: ['react', 'prettier'],
  ignorePatterns: ['/src/opencv'],
  rules: {
    quotes: ['error', 'single'],
    'prettier/prettier': [
      'error',
      {
        trailingComma: 'es5',
        singleQuote: true,
        printWidth: 80,
      },
    ],
  },
};
