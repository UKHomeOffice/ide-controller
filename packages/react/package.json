{
  "name": "@ide-controller/react",
  "version": "1.6.0",
  "private": true,
  "homepage": "./",
  "author": "Viable Data",
  "scripts": {
    "dev": "BROWSER=none CI=true react-scripts start",
    "build": "react-scripts build",
    "test": "react-scripts test --watchAll=false",
    "test:watch": "react-scripts test",
    "eject": "react-scripts eject",
    "lint": "npx eslint --ext .js,.jsx .",
    "lint:fix": "eslint --ext .js,.jsx . --fix"
  },
  "browserslist": {
    "production": [
      ">0.2%",
      "not dead",
      "not op_mini all"
    ],
    "development": [
      "last 1 chrome version",
      "last 1 firefox version",
      "last 1 safari version"
    ]
  },
  "dependencies": {
    "@tensorflow-models/posenet": "^2.2.1",
    "@tensorflow/tfjs": "^2.7.0",
    "govuk-frontend": "^3.8.0",
    "prop-types": "^15.7.2",
    "react": "^16.13.1",
    "react-dom": "^16.13.1",
    "react-scripts": "^3.4.3"
  },
  "devDependencies": {
    "@testing-library/jest-dom": "^4.2.4",
    "@testing-library/react": "^9.3.2",
    "@testing-library/user-event": "^7.1.2",
    "eslint-config-airbnb": "^18.2.0",
    "eslint-config-prettier": "^6.11.0",
    "eslint-plugin-import": "^2.22.1",
    "eslint-plugin-prettier": "^3.1.4",
    "eslint-plugin-react": "^7.21.5",
    "jest-canvas-mock": "^2.3.0",
    "node-sass": "^4.14.1",
    "prettier": "^2.1.1"
  }
}
