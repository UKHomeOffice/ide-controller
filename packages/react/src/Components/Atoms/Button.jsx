// Global imports
import PropTypes from 'prop-types';
import React from 'react';

const Button = ({ buttonVariant, onClick, children, disabled, className }) => {
  return (
    <button
      className={`govuk-button font--19pt govuk-button--${buttonVariant} ${className}`}
      data-module="govuk-button"
      type="button"
      onClick={onClick}
      disabled={disabled}
      data-testid="atoms-button"
    >
      {children}
    </button>
  );
};

Button.propTypes = {
  children: PropTypes.string.isRequired,
  buttonVariant: PropTypes.string,
  onClick: PropTypes.func,
  disabled: PropTypes.bool,
  className: PropTypes.string,
};

Button.defaultProps = {
  buttonVariant: 'secondary',
  onClick: null,
  disabled: false,
  className: '',
};

export default Button;
