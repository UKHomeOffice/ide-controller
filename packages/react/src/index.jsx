// Global imports
import React from 'react';
import ReactDOM from 'react-dom';
import { initAll } from 'govuk-frontend';
import 'govuk-frontend/govuk/all.scss';

// Local imports
import App from './App';
import './Components/Style/atoms.scss';
import './Components/Style/global.scss';
import './Components/Style/molecules.scss';
import './helpers/ipcMainEvents';

document.body.className = document.body.className
  ? `${document.body.className} js-enabled`
  : 'js-enabled';

initAll();

ReactDOM.render(
  <React.StrictMode>
    <App />
  </React.StrictMode>,
  document.getElementById('root')
);
