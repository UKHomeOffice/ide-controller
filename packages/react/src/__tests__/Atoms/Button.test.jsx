// Global imports
import React from 'react';
import { render, fireEvent, screen } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';

// Local imports
import Button from '../../Components/Atoms/Button';

describe('<Button />', () => {
  it('renders and match the snapshot', () => {
    const onClick = jest.fn();
    const { asFragment } = render(
      <Button buttonVariant="secondary" disabled={false} onClick={onClick}>
        Test
      </Button>
    );

    expect(asFragment).toMatchSnapshot();
  });

  it('calls onClick props when clicked', () => {
    const onClick = jest.fn();
    render(<Button onClick={onClick}>Click Me</Button>);
    fireEvent.click(screen.getByTestId('atoms-button'));

    expect(onClick).toHaveBeenCalledTimes(1);
  });

  it('is disabled when disabled prop equals true', () => {
    render(<Button disabled>Click Me</Button>);
    const button = screen.getByTestId('atoms-button');

    expect(button).toBeDisabled();
  });

  it('has properly className', () => {
    render(<Button className="secondary">Click Me</Button>);
    const button = screen.getByTestId('atoms-button');

    expect(button).toHaveClass('govuk-button--secondary');
  });

  it('has properly text content', () => {
    const text = 'Click Me';
    render(<Button>{text}</Button>);
    const button = screen.getByTestId('atoms-button');

    expect(button).toHaveTextContent(text);
  });
});
