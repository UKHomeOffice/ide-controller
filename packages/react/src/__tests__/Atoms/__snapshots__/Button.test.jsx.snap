// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> renders and match the snapshot 1`] = `[Function]`;
